### PR TITLE
Try and fix Sentry errors

### PR DESCRIPTION
**Pull request by theefer**
Sourcemaps are better now though not 100% reliable still possibly.

### CODE REVIEW

File: kahuna/public/js/image/controller.js [hunk 1]
```diff
@@ -194,10 +194,12 @@ image.controller('ImageCtrl', [
...
 
         const freeImageDeleteFailListener = $rootScope.$on('image-delete-failure', (err, image) => {
-            if (err.body && err.body.errorMessage) {
+            if (err && err.body && err.body.errorMessage) {
                 $window.alert(err.body.errorMessage);
             } else {
-                $window.alert(`Failed to delete image ${image.data.id}`);
+                // Possibly not receiving a proper image object sometimes?
+                const imageId = image && image.data && image.data.id || 'Unknown ID';
+                $window.alert(`Failed to delete image ${imageId}`);
```
Comment (theefer) [inline kahuna/public/js/image/controller.js]: Possibly https://beta.getsentry.com/the-guardian/the-grid/group/81091607/events/3770690434/

File: kahuna/public/js/usage-rights/usage-rights-editor.js [hunk 2]
```diff
@@ -200,7 +200,9 @@ usageRightsEditor.controller(
     }
 
     function uiError(error) {
-        ctrl.error = error.body.errorMessage;
+        // ♫ Very superstitious ♫
+        ctrl.error = error && error.body && error.body.errorMessage ||
+            'Unexpected error';
```
Comment (theefer) [inline kahuna/public/js/usage-rights/usage-rights-editor.js]: https://beta.getsentry.com/the-guardian/the-grid/group/81090041/

File: kahuna/public/js/main.js [hunk 3]
```diff
@@ -125,7 +125,7 @@ kahuna.run(['$log', '$rootScope', 'mediaApi', function($log, $rootScope, mediaAp
                     authAndRedirect(loginUriTemplate);
                 } else {
                     // Couldn't extract a login URI, die noisily
-                    throw error;
+                    throw new Error('Failed to redirect to auth, no login URI found');
```
Comment (theefer) [inline kahuna/public/js/main.js]: Unlikely but possibly https://beta.getsentry.com/the-guardian/the-grid/group/80626909/ (?)

Comment (akash1810) [inline kahuna/public/js/main.js]: worth doing `$log.info(error)` too?

Comment (theefer) [inline kahuna/public/js/main.js]: Errors thrown in a Promise chain should be surfaced on the window by Angular anyway, so I think that would be superfluous.

Comment (theefer) [inline kahuna/public/js/main.js]: Note that we normally don't throw errors in Promise chains, we `$q.reject` instead. We should only throw if we want it to be surfaced noisily in our errors.

**Comment by akash1810** [comment]
:+1:

**Pull request merged**